File: java/observer/MiddleObserver.java
public class MiddleObserver extends Observer {
    public MiddleObserver(Car car){
        super(car);
    }
    public void update(){
        int pos = getCar().getPosition();
        if (pos == 0)
        {
            System.out.println("run straight");
        }
    }
}

File: cpp/observer/MiddleObserver.cpp
#include "MiddleObserver.h"

MiddleObserver::MiddleObserver(Car *car) : Observer(car) {}

void MiddleObserver::update()
{
    int pos = getCar()->getPosition();
    if (pos == 0)
    {
        cout << "run straight" << endl;
    }
}